src/TestProblem/Hydro/Zeldovich/Init_TestProb_Zeldovich.cpp

```diff
@@ -0,0 +1,537 @@
+      Aux_Error( ERROR_INFO, "OPT__INIT != FUNCTION (1) or RESTART (2) for this test !!\n" );
+
+#  ifndef FLOAT8
+   Aux_Error( ERROR_INFO, "FLOAT8 must be enabled !!\n" );
+#  endif
+
+#  ifndef SUPPORT_GSL
+   Aux_Error( ERROR_INFO, "SUPPORT_GSL must be enabled !!\n" );
+#  endif
+
+#  ifdef PARTICLE
+   if ( amr->Par->Init != PAR_INIT_BY_FUNCTION )
+      Aux_Error( ERROR_INFO, "PAR_INIT != FUNCTION (1) for this test !!\n" );
+#  endif
+
+
+   if ( MPI_Rank == 0 )    Aux_Message( stdout, "   Validating test problem %d ... done\n", TESTPROB_ID );
+
+} // FUNCTION : Validate
+
+
+
+#if ( MODEL == HYDRO )
+//-------------------------------------------------------------------------------------------------------
+// Function    :  SetParameter
+// Description :  Load and set the problem-specific runtime parameters
+//
+// Note        :  1. Filename is set to "Input__TestProb" by default
+//                2. Major tasks in this function:
+//                   (1) load the problem-specific runtime parameters
+//                   (2) set the problem-specific derived parameters
+//                   (3) reset other general-purpose parameters if necessary
+//                   (4) make a note of the problem-specific parameters
+//
+// Parameter   :  None
+//
+// Return      :  None
+//-------------------------------------------------------------------------------------------------------
+void SetParameter()
+{
+
+   if ( MPI_Rank == 0 )    Aux_Message( stdout, "   Setting runtime parameters ...\n" );
+
+
+// (1) load the problem-specific runtime parameters
+   const char FileName[] = "Input__TestProb";
+   ReadPara_t *ReadPara  = new ReadPara_t;
+
+// (1-1) add parameters in the following format:
+// --> note that VARIABLE, DEFAULT, MIN, and MAX must have the same data type
+// --> some handy constants (e.g., Useless_bool, Eps_double, NoMin_int, ...) are defined in "include/ReadPara.h"
+// ********************************************************************************************************************************
+// ReadPara->Add( "KEY_IN_THE_FILE",   &VARIABLE,               DEFAULT,      MIN,              MAX               );
+// ********************************************************************************************************************************
+   ReadPara->Add( "Gas_Par_Setup",     &Gas_Par_Setup,          1,            1,                2                 );
+   ReadPara->Add( "n_Pert_Wave_Len",   &n_Pert_Wave_Len,        2,            1,                NoMax_int         );
+   ReadPara->Add( "zc_Collapse",       &zc_Collapse,            1.0,          0.0,              NoMax_double      );
+   ReadPara->Add( "GasTemp_Init",      &GasTemp_Init,           100.0,        0.0,              NoMax_double      );
+   ReadPara->Add( "NPar_X",            &NPar_X,                 NX0_TOT[0],    0,                NoMax_int         );
+
+   ReadPara->Read( FileName );
+
+   delete ReadPara;
+
+
+// (1-2) set the default values
+
```

Comment:
- [ ] Set `zi_Initial = 1.0/Time - 1.0;` here.
- [ ] Do not set `zi_Initial` redundantly in `Par_Init_ByFunction_Zeldovich()` and `SetGridIC()`.